### PR TITLE
Fixate Driver Tests

### PR DESCRIPTION
**Pull request by Jasper-Harvey0**
Made a start on the driver tests. 

Definitely going to be some silly stuff in here that needs changing

### CODE REVIEW

File: test/drivers/test_keysight_33500b_pytest.py [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+
+@pytest.mark.drivertest
+def test_open_funcgen():
+    import fixate.drivers.funcgen
+    funcgen = fixate.drivers.funcgen.open()
+    assert funcgen, "Could not open function generator"
+
+@pytest.mark.drivertest
+def test_sin(funcgen):
+    funcgen.channel1.waveform.square()
```
Comment (John2202W) [inline test/drivers/test_keysight_33500b_pytest.py:21]: Doubled up?

File: test/drivers/test_keysight_33500b_pytest.py [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+
+    assert re.search("value clipped to upper limit", str(excinfo.value))
+
+    # Lower Limits
+    funcgen.channel1.vrms(354e-6)
+    assert float(funcgen.instrument.query("SOUR1:VOLT?")) == pytest.approx(354e-6)
+    with pytest.raises(InstrumentError) as excinfo:
+        funcgen.channel1.vrms(353e-6)
+
+    assert re.search("value clipped to lower limit", str(excinfo.value))
```
Comment (John2202W) [inline test/drivers/test_keysight_33500b_pytest.py:92]: Should break these up as individual tests and parameterise each of the functions and input values

File: test/drivers/test_keysight_33500b_pytest.py [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+from fixate.config import load_config
+load_config() # Load fixate config file
+
+# rewirte of "test_keysight_33500b.py" into pytest framework
+
+# TODO: Use fixture setup and taredown to clean up test code.
+# TODO: Separate early tests with multiple asserts into tests with one assert each. Make use of mark parameterised
+
+
+@pytest.mark.drivertest
```
Comment (John2202W) [inline test/drivers/test_keysight_33500b_pytest.py:13]: Like the idea of these marks for skipping

**Comment by Jasper-Harvey0** [comment]
Workflow checks pass on my fork now.

**Comment by daniel-montanari** [comment]
@Jasper-Harvey0 are you able to fix this conflict please?
Other than that is everything else ready for merging?

**Comment by John2202W** [comment]
Might need to test these out on the new Siglent F/W aswell - in particular the over current/voltage tests that are expecting certain system error strings to be returned

File: test/conftest.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import pytest
+import fixate.drivers.funcgen
+import fixate.drivers.dso
+from drivers.J413 import dm
```
[the remark this answers is not in the document]
Comment (daniel-montanari) [reply test/conftest.py]: This should hopefully work with the new driver system that doesn't crash on import

**Comment by daniel-montanari** [comment]
Should these drivers be marked with drivertest?
```
test\drivers\test_keysight_33500b.py sssssssssssssssssssssssssssssssssss [ 62%]
sssssssssssssssssssssssss                                                [ 75%]
test\drivers\test_rigol_dg1022.py ssssssssssssssssssssssssssssssssssssss [ 93%]
sssssssssssss                                                            [100%]
```
Otherwise I'll merge if no one else has any issues

**Comment by Jasper-Harvey0** [comment]
I removed that old Keysight test written in unittest. Should be all good now.

**Comment by Jasper-Harvey0** [comment]
@jcollins1983 Can we get this merged before merging the driver addition? That way I can pull this into my driver addition and add tests for the new DMM.

**Review by jcollins1983 — COMMENTED** [review]
Just a few minor comments.

**Comment by Jasper-Harvey0** [comment]
I have made those changes

**Pull request closed without merging**